### PR TITLE
feat: support legacy transaction in EOA

### PR DESCRIPTION
**Pull request by Flydexo**
Added support for legacy transactions, see reference here:
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
https://ethereum-magicians.org/t/eip-2718-typed-transaction-envelope/4355

Time spent on this PR: 3 days

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Legacy Transactions are refused by default

## What is the new behavior?

Legacy transactions are handled, verified.

### CODE REVIEW

File: src/kakarot/accounts/eoa/aa/library.cairo [hunk 1]
```diff
@@ -165,15 +207,38 @@ namespace ExternallyOwnedAccount {
...
+        tx_type: felt,
+        keccak_ptr: felt*,
+        keccak_ptr_start: felt*,
+    ) -> (is_valid: felt) {
         alloc_locals;
-        let (keccak_ptr: felt*) = alloc();
-        local keccak_ptr_start: felt* = keccak_ptr;
-        with keccak_ptr {
-            verify_eth_signature_uint256(msg_hash=msg_hash, r=r, s=s, v=v, eth_address=eth_address);
+        if (tx_type == 2) {
```
Comment (ClementWalter) [inline src/kakarot/accounts/eoa/aa/library.cairo:219]: so `type = 2` introduces only a shift (3) in the indexes `V_IDX`, `R_IDX` and `S_IDX`. We could better just compute the correct indices and avoid the `if/else` and duplication.

```
let shift = tx_type / 2 * 3
let v_idx = V_IDX - shift
...
```

**Comment by TotalPizza** [comment]
In general there are some more things we need to validate besides the signature:
Correct nonce, gasLimit is withing valid range, and a few more.
But it's probably best we do this with the TLV-Hackathon pr and not this one.

**Comment by Flydexo** [comment]
> In general there are some more things we need to validate besides the signature: Correct nonce, gasLimit is withing valid range, and a few more. But it's probably best we do this with the TLV-Hackathon pr and not this one.

Yes, agree was starting for gas but the testing library doesn't allow spending gas on tx

**Review by Eikix — CHANGES_REQUESTED** [review]
What is the status of this PR after the TLV hacker house? 
@Flydexo @jobez @TotalPizza did anyone "fork" Flydexo's branch to complete this PR? Or is it still "relevant"?

**Comment by jobez** [comment]
> What is the status of this PR after the TLV hacker house? @Flydexo @jobez @TotalPizza did anyone "fork" Flydexo's branch to complete this PR? Or is it still "relevant"?

I believe it is still relevant, as it extends the fork we were working with to add legacy support.

**Pull request closed without merging**